scenario_gym/scenario_gym.py

```diff
@@ -307,13 +308,59 @@ def close(self) -> None:
 
     def get_metrics(self) -> Dict[str, Any]:
         """Get the current metric states."""
-        values = {}
+        merged_dict = {}
+        extra_metrics = {}
+
+        # Iterate through each metric
         for metric in self.metrics:
-            value = metric.get_state()
-            if isinstance(value, dict):
-                for k, v in value.items():
-                    if isinstance(k, str):
-                        values[f"{metric.name}_{k}"] = v
-            elif value is not None:
-                values[metric.name] = value
-        return values
+            value = (
+                metric.get_state()
```

Comment:
Are there some extra assumptions about the keys of the nested dicts? Are they floats?

Reply:
Hi @hamishs! Apologies for the extremely delayed reply.

Yes, the keys of the nested dicts are assumed to be floats since the round() function acts on them. However, this does not affect the metrics developed on your end (such as EgoAvgSpeed), and simply wraps the output in a dictionary with the key "Lagging Metrics Post-Runtime". 

For example, if I only call EgoAvgSpeed and EgoMaxSpeed (default implemented metrics with float output) and CollisionTimestamp (another float output) like so:
![image](https://github.com/user-attachments/assets/7ed01ef7-e8ff-471a-a69d-38b0beacc1d7)
We get an output like this:
![image](https://github.com/user-attachments/assets/4a1bf2dd-7b7a-4438-bb02-a1057a60081c)

Reply:
Hi @KethanR, okay I understand. Thank you for taking the time to explain and to work on the PR however I think that this is a bit complicated for the public API which we would prefer to keep simple.

Can I suggest that you implement this as a standalone function or as part of a subclass of `ScenarioGym` in your codebase?